features/step_definitions/basic_steps.rb

```diff
@@ -22,23 +22,25 @@
 
 Given('We have the following recipes:') do |table|
   table.hashes.each do |recipe|
-    if recipe[:user] && recipe[:image]
-      user = User.find_by email: recipe[:user]
-      recipe_hash = recipe.except('user', 'image')
-      new_recipe = create(:recipe, recipe_hash.merge(user: user))
+    if recipe[:user]
+      recipe[:user] = User.find_by email: recipe[:user]
+    end
+
+    if recipe["forked from"]
+      recipe[:original_recipe_id] = Recipe.find_by(title: recipe["forked from"]).id if recipe["forked from"] != ""
+      recipe.delete("forked from")
+    end
+
+    if recipe[:category]
+      recipe[:category] = Category.find_by name: recipe[:category]
+    end
+
+    new_recipe = create(:recipe, recipe.except('image'))
+
+    if recipe[:image]
```

Comment:
Not sure if this refactoring is in scope. What do we achieve with this change?

Reply:
I was using the same function in test with a new attribute `forked from`
When merging had to combine the two function into one, and making sure it works for all tests.